mmseg/models/backbones/strideformer.py

```diff
@@ -0,0 +1,765 @@
+
+        q = self.to_q(x)  # [B, nhead*dim, H, W]
+        k = self.to_k(x)
+        v = self.to_v(x)
+
+        qkv = torch.cat([q, k, v], dim=1)
+        qkv = self.act(self.dwconv(qkv))
+        qkv = self.pwconv(qkv)
+
+        qrow = self.pos_emb_rowq(q.mean(-1)).reshape(
+            [B, self.num_heads, -1, H]).permute(
+            (0, 1, 3, 2))  # [B, nhead, H, dim]
+        krow = self.pos_emb_rowk(k.mean(-1)).reshape(
+            [B, self.num_heads, -1, H])  # [B, nhead, dim, H]
+        vrow = v.mean(-1).reshape([B, self.num_heads, -1, H]).permute(
+            [0, 1, 3, 2])  # [B, nhead, H, dim*attn_ratio]
+
+        attn_row = torch.matmul(qrow, krow) * self.scale  # [B, nhead, H, H]
+        attn_row = nn.functional.softmax(attn_row, dim=-1)
+
+        xx_row = torch.matmul(attn_row, vrow)  # [B, nhead, H, dim*attn_ratio]
+        xx_row = self.proj_encode_row(
+            xx_row.permute([0, 1, 3, 2]).reshape([B, self.dh, H, 1]))
+
+        # squeeze column
+        qcolumn = self.pos_emb_columnq(q.mean(-2)).reshape(
+            [B, self.num_heads, -1, W]).permute([0, 1, 3, 2])
+        kcolumn = self.pos_emb_columnk(k.mean(-2)).reshape(
+            [B, self.num_heads, -1, W])
+        vcolumn = torch.mean(v, -2).reshape(
+            [B, self.num_heads, -1, W]).permute([0, 1, 3, 2])
+
+        attn_column = torch.matmul(qcolumn, kcolumn) * self.scale
+        attn_column = nn.functional.softmax(attn_column, dim=-1)
+
+        xx_column = torch.matmul(attn_column, vcolumn)  # B nH W C
+        xx_column = self.proj_encode_column(
+            xx_column.permute([0, 1, 3, 2]).reshape([B, self.dh, 1, W]))
+
+        xx = torch.add(xx_row, xx_column)  # [B, self.dh, H, W]
+        xx = torch.add(v, xx)
+
+        xx = self.proj(xx)
+        xx = self.sigmoid(xx) * qkv
+        if self.stride_attention:
+            xx = F.interpolate(xx, size=(H_ori, W_ori), mode='bilinear')
+
+        return xx
+
+
+class MLP(nn.Module):
+    def __init__(self,
+                 in_features,
+                 hidden_features=None,
+                 out_features=None,
+                 act_layer=None,
+                 conv_cfg=dict(type='Conv'),
+                 drop=0.):
+        super().__init__()
+        out_features = out_features or in_features
+        hidden_features = hidden_features or in_features
+        self.fc1 = Conv2DBN(in_features, hidden_features)
+        self.dwconv = build_conv_layer(conv_cfg, hidden_features, hidden_features, 3, 1, 1, groups=hidden_features)
+        self.act = act_layer
+        self.fc2 = Conv2DBN(hidden_features, out_features)
+        self.drop = build_dropout(dict(type='Dropout', drop_prob=drop))
+
+    def forward(self, x):
+        x = self.fc1(x)
+        x = self.dwconv(x)
+        x = self.act(x)
+        x = self.drop(x)
+        x = self.fc2(x)
+        x = self.drop(x)
+        return x
+
+
+class Fusion_block(nn.Module):
+    def __init__(self, inp, oup, embed_dim, activations=None) -> None:

```

Comment:
Might rename to `FusionBlock`.

Reply:
And the names of parameters are not readable, such as `inp`, `oup`. It suggests changing to names more readable.